cypress/e2e/gui/homepage_gui_tests.cy.js

```diff
@@ -0,0 +1,37 @@
+
+describe('Homepage GUI validations', () => {
+
+    // Constants used across multiple tests
+    const today = new Date(2024, 6, 9) // July 9, 2024
+    const formattedDate = today.toDateString()
+
+    beforeEach(() => {
+        cy.visit('/')
+        cy.clock(today.getTime())
+    })
+
+    it('Homepage renders the header with an h1 and theme toggle*', () => {
+
+        cy.contains('h1', 'EngageSphere')
+            .should('be.visible')
+
+        cy.get('#theme-toggle-button')
+            .should('be.visible')
+    })
+    it('Homepage shows the default greeting (i.e., Hi there! ...) *', () => {
+
+        cy.get('[data-testid="table"].table-container')
```

Comment:
What about something a little simpler, like...

```suggestion
        cy.contains('p', 'Hi there! It is now Tue Jul 9 2024.').should('be.visible')
```

Reply:
I think your suggestion is much cleaner and improves the readability of the test. In my mind, I always try to be as specific as possible when choosing the accessibility element aimed at making sure it is as unique as possible. e.g. what if `'Hi there! It is now Tue Jul 9 2024.'` was assigned to the wrong <p> element? Hence, I tried to limit the chance of this happening by using very unique elements, although in this case using a class is probably not unique enough.